week3/homework/hackyourtemperature/server.js

```diff
@@ -0,0 +1,37 @@
+const express = require("express");
+const fetch = require("node-fetch");
+const API_KEY = require("./sources/keys.json").API_KEY;
+const app = express();
+const exphbs = require("express-handlebars");
+const port = process.env.PORT || 3000;
+
+app.set("view engine", "handlebars");
+app.engine("handlebars", exphbs({ defaultLayout: false }));
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
+
+app.post("/weather", async (req, res) => {
+  const cityName = req.body.cityName;
+  if (typeof cityName === "undefined" || cityName === "") {
+    res.status(404).render("index", { weatherText: "City is not found!" });
```

Comment:
If the city is not found you are showing an error message to the user, that's perfect; however you are showing it in the same "format" as a successful response, to be more future proof it would be nice to have a different objects for errors like 
```suggestion
    res.status(404).render("index", { error: "City is not found!" });
```
In this case you would have to also update your html template to show the error (as an error).